### PR TITLE
show condition status on component list page

### PR DESCRIPTION
**Pull request by sahil143**
## Fixes 

https://issues.redhat.com/browse/HAC-504

## Description

Show Conditions status on UI after a component is created

## Type of change

- [ ] Feature
- [x] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Screen shots / Gifs for design review 

image
image

## Browser conformance: 

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Review by rottencandy — APPROVED** [review]
/lgtm

**Comment by itsptk** [comment]
Once a component has been created successfully, does this status have any purpose? Or is it always green then?

When does this become "Component updated"? Can those updates ever fail?

**Comment by rohitkrai03** [comment]
@sahil143 ^^

**Comment by sahil143** [comment]
>Once a component has been created successfully, does this status have any purpose? Or is it always green then?

@itsptk if there is an error from HAS it is reflected in the status conditions after the component is created. It's not always green if you notice in the first screenshot, the component resource was created but was unable to generate gitops resource.

> When does this become "Component updated"? Can those updates ever fail?

It becomes "Component Updated` once we edit the component resource. If Create can fail then I assumed Update can also fail in similar manner.

**Comment by itsptk** [comment]
UX is intending to consider statuses around the application view and it's components holistically via https://issues.redhat.com/browse/DTUX-1260 for M7. My inclination would be to wait for that designs before adding anything around this status to the UI.

If we need to implement something sooner than that, I'd suggest only showing the failure case with a message like "component creation/update failed" and not show anything for the success case.

File: src/utils/common-utils.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { K8sResourceCommon } from '@openshift/dynamic-plugin-sdk-utils';
+import { ResourceStatusCondition } from '../types';
+
+export const getConditionForResource = <R extends K8sResourceCommon>(
```
[the remark this answers is not in the document]
Comment (sahil143) [reply src/utils/common-utils.ts:4]: yes, It should work for HAS resources

**Review by rottencandy — APPROVED** [review]
Shows condition properly
/lgtm

**Comment by sahil143** [comment]
image
image
image

**Review by rohitkrai03 — APPROVED** [review]
/approve

**Pull request merged**